### PR TITLE
First working prototype of ITS3 digitiser

### PR DESCRIPTION
**Pull request by akalweit**
Dear @shahor02,
we had to modify common code in some places. We kept the default behaviour, but please have a look.

Thanks,
Max and Alex

### CODE REVIEW

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/SimTraits.h
Comment (shahor02) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/SimTraits.h]: This is already in dev.

File: Detectors/ITSMFT/common/simulation/src/AlpideSimResponse.cxx
Comment (shahor02) [inline Detectors/ITSMFT/common/simulation/src/AlpideSimResponse.cxx]: please use `LOG(ERROR)`

File: Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h
Comment (shahor02) [inline Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h]: in the SegmentationAlpide the guardring is the part of the sensor (epitaxial layer) and is accounted in coordinates transformation, as you also did

**Review by shahor02 — CHANGES_REQUESTED** [review]
Hi @akalweit 
looks fine, just minor comments + CI complains on untested macro.

File: Detectors/ITSMFT/common/simulation/src/AlpideSimResponse.cxx
Comment (mpuccio) [reply Detectors/ITSMFT/common/simulation/src/AlpideSimResponse.cxx]: Thanks for spotting this, it was a debug cout that we did not remove!

File: Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h
Comment (akalweit) [reply Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h]: We are not sure if we understand you correctly, but we just copied this comment from SegmentationAlpide. So if it is wrong here, it isalso wrong there.

**Comment by akalweit** [comment]
We should have fixed the requested changes. The only remaining comment is about the documentation.

**Comment by mpuccio** [comment]
Hi @shahor02, CI is green :) Shall we still see how to fix the comment on the segmentation?

File: Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h
Comment (shahor02) [reply Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h]: OK, I did not realize that I am commenting on my comment, then the logic/math for pixel<->local coord formally is correct. But I see that you are defining the PassiveEdgeReadOut in a same way as for the Alpide, i.e. it is a dead zone along Z (columns count). Isn't cylindrical layer readout done from the side(s)? If so, then the meaning of top/bottom and side guardrings should be swapped as well the meaning of row and column.

**Review by shahor02 — COMMENTED** [review]
@mpuccio see new comment about the segmentation. Let me know if you want me to merge it as it is now and address this problem later.

File: Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h
Comment (akalweit) [reply Detectors/Upgrades/IT3/base/include/ITS3Base/SegmentationSuperAlpide.h]: Thanks for noticing! We have set the passive edge to zero for the time being. We still do not know how the passive areas will look like in the final detector and we leave this detail to future implementations.

**Comment by akalweit** [comment]
Thanks @shahor02 ! As soon as the CI is green, we are ready to merge. Max &Alex

**Review by shahor02 — DISMISSED** [review]
OK, but please apply clang-format for the CI to complete, will merge after it.

**Comment by akalweit** [comment]
Yes, CLANG is fixed now. Many thanks!

**Review by shahor02 — APPROVED** [review]

**Pull request merged**
